### PR TITLE
NIFI-8632: Using available port in StandardOidcIdentityProviderGroovy…

### PR DESCRIPTION
**Pull request by gresockj**
…IT.groovy

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Converts StandardOidcIdentityProviderGroovyTest to StandardOidcIdentityProviderGroovyIT since the test attempts to connect to a mock HTTP server using an actual port.  Additionally, uses an available port instead of assuming port 443 is available, when running it as an IT.

This specific test has the possibility of receiving a Connection Timeout instead of Connection Refused.  In order to support this, an upgrade to junit 4.3.12 was needed, since this version addressed some unit test timeout issues.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy [hunk 1]
```diff
@@ -58,17 +58,29 @@ import org.junit.runners.JUnit4
...
+    private static final Logger logger = LoggerFactory.getLogger(StandardOidcIdentityProviderGroovyIT.class)
 
     private static final Key SIGNING_KEY = new Key(id: 1, identity: "signingKey", key: "mock-signing-key-value")
 
+    private static int getAvailablePort() throws IOException {
+        SocketChannel socket = SocketChannel.open()
+        socket.setOption(StandardSocketOptions.SO_REUSEADDR, true)
+        socket.bind(new InetSocketAddress("localhost", 0))
+        return socket.socket().getLocalPort()
+    }
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy:74]: The `NetworkUtils.getAvailableTcpPort()` in `nifi-commons` performs this function and also closes the socket after returning.  Recommend leveraging that class for this unit test.

**Review by exceptionfactory — COMMENTED** [review]
Thanks for addressing this issue @gresockj!  See the note on leveraging the existing `NetworkUtils.getAvailableTcpPort()`, otherwise looks straightforward.

File: nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy (hunk 1 above)
Comment (gresockj) [reply nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy:74]: Unfortunately, Groovy doesn't like the try-with-resources call in that method.  However, this helped me track down the Connection timeout issue!  I just needed to close the socket.

File: nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy [hunk 2]
```diff
@@ -21,11 +21,7 @@ import com.nimbusds.jwt.JWTClaimsSet
 import com.nimbusds.jwt.PlainJWT
 import com.nimbusds.oauth2.sdk.AuthorizationCode
 import com.nimbusds.oauth2.sdk.AuthorizationCodeGrant
-import com.nimbusds.oauth2.sdk.auth.ClientAuthentication
-import com.nimbusds.oauth2.sdk.auth.ClientAuthenticationMethod
-import com.nimbusds.oauth2.sdk.auth.ClientSecretBasic
-import com.nimbusds.oauth2.sdk.auth.ClientSecretPost
-import com.nimbusds.oauth2.sdk.auth.Secret
+import com.nimbusds.oauth2.sdk.auth.*
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy:24]: Did your IDE automatically condense these imports?  Recommend reverting to explicit imports.

Comment (gresockj) [reply nifi-registry/nifi-registry-core/nifi-registry-web-api/src/test/groovy/org/apache/nifi/registry/web/security/authentication/oidc/StandardOidcIdentityProviderGroovyIT.groovy:24]: Yikes, it sure did, good catch.

**Review by exceptionfactory — APPROVED** [review]
Thanks @gresockj, looks good!

**Pull request closed without merging**